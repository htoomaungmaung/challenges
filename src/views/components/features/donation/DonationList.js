// @flow
import React, { useState } from 'react';
import styled from 'styled-components';
import Charity from './charity/Charity';

import {ComponentErrorBoundary as ErrorBoundary} from '../../../utils/errorBoundary';

// aspect ratio 16:9 + extra 20% for displaying info
const maxWidth = 800;
const minHeight = 540;
const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-evenly;
`;

const Card = styled.div`
  flex: 50%;
  margin: 2rem;
  max-width: ${maxWidth}px;
  min-width: ${maxWidth}px;
  min-height: ${minHeight};
  max-height: ${minHeight};
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  position: relative;
  box-shadow: 0 2px 8px rgba(0,0,0,0.26);
  border-radius: 5px;
  @media all and (max-device-width: ${maxWidth}px)  {
    width: 100%;
    height: auto;
    min-width: 400px;
    min-height: 281px;
  }
  @media all and (max-width: ${maxWidth}px)  {
    width: 100%;
    height: auto;
    min-width: 400px;
    min-height: 281px;
  }
`;

type charity = {
  name: string,
  totalDonation: number,
  currency: string,
  id: any,
  image: any
}

type Props = {
  charities: Array<charity>,
  paymentHandler: any,
  
}

const DonationList = (props: Props) => {
  let donationList; 
    if(props.charities){
      donationList = props.charities.map((item, i) => (
        <Card key={i}>        
              <ErrorBoundary>
                    <Charity 
                    charityId = {item.id}
                    name={item.name} 
                    totalDonation={item.totalDonation} 
                    currency={item.currency}
                    image={item.image}
                    paymentHandler={props.paymentHandler} />         
              </ErrorBoundary>
          </Card>
      ));
    }
    return (
      <Container>        
        {donationList}        
      </Container>
    );
}


export default DonationList;