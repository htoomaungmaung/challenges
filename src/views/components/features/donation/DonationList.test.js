import React from 'react';
import { shallow } from 'enzyme';
import DonationList from './DonationList';
import renderer from 'react-test-renderer';

describe('<DonationList />', () => {
  const props={
    charities: [{
      name: "",
      totalDonation: 0,
      currency: "",
      id: ""
    }],
    paymentHandler: jest.fn()
  }

  it('<DonationList /> renders without crashing', () => {
    const wrapper = shallow(<DonationList {...props} />);
    expect(wrapper.exists()).toBe(true);
  });

  it('<DonationList /> renders without props', () => {
    const wrapper = shallow(<DonationList />);
    expect(wrapper.exists()).toBe(true);
  });

  it('<DonationList /> snapshot test', () => {
    const tree = renderer.create(<DonationList {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
    it('<DonationList /> snapshot test without props', () => {
    const tree = renderer.create(<DonationList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
