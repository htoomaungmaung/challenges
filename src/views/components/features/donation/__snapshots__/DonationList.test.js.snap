// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DonationList /> <DonationList /> snapshot test 1`] = `
<div
  className="sc-dnqmqq iTnEek"
>
  <div
    className="sc-iwsKbI jcKscV"
  >
    <div>
      <div
        className="sc-ifAKCX cJLgyY"
      >
        <img
          src="images/undefined"
          style={
            Object {
              "zIndex": "2",
            }
          }
        />
        <div
          className="sc-bdVaJa isMeQX"
        >
          <p>
            Select the amount to donate 
            
             
          </p>
          <div
            className="sc-bwzfXH cRuqQG"
          >
            <label>
              <input
                checked={true}
                onChange={[Function]}
                type="radio"
                value={10}
              />
              10
            </label>
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value={20}
              />
              20
            </label>
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value={50}
              />
              50
            </label>
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value={100}
              />
              100
            </label>
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value={500}
              />
              500
            </label>
          </div>
          <button
            className="sc-htpNat XpVwB"
            onClick={[Function]}
          >
            Pay
          </button>
        </div>
      </div>
      <div
        className="sc-EHOje hoBdWw"
      >
        <div
          className="sc-bZQynM dmujVc"
        >
          <p>
            
          </p>
          
        </div>
        <div
          className="sc-gzVnrw ihGDdT"
        >
          <button
            className="sc-htoDjs iVuBhZ"
            onClick={[Function]}
          >
            Donate
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DonationList /> <DonationList /> snapshot test without props 1`] = `
<div
  className="sc-dnqmqq iTnEek"
/>
`;
